pkg/snapshot/snapshotter/snapshotter.go

```diff
@@ -736,19 +736,24 @@ func (ssr *Snapshotter) resetFullSnapshotTimer() error {
 	return nil
 }
 
+// checkSnapstoreSecretUpdate checks if the snapstore secret has been updated
 func (ssr *Snapshotter) checkSnapstoreSecretUpdate() bool {
```

Comment:
This method is called from `TakeDeltaSnapshot`. Why is not called when taking full snapshot? What happens when a secret has changed after taking delta but before a full snapshot is triggered?
Essentially whenever you take a snapshot, first thing that you do is create a SnapStore.

Reply:
Please refer to [this](https://github.com/gardener/etcd-backup-restore/pull/670#issuecomment-1811747574) discussion thread for the explanation.
@unmarshall

Reply:
i do not get the reasoning. It is a simple change.

Reply:
Will make the necessary change for full snapshots.